VideoLocker/src/main/java/org/edx/mobile/http/Api.java

```diff
@@ -1223,6 +1221,12 @@ public RegistrationDescription getRegistrationDescription() throws Exception {
     }
 
     public boolean enrollInACourse(String courseId, boolean email_opt_in) throws Exception {
+        // uses auth headers
+        Bundle headers = getAuthHeaders();
+
+        // this API endpoint also requires CSRFToken, so add it
+        headers = setCookieHeaders(headers);
```

Comment:
Looking closer, it looks ``like getAuthHeaders`` actually *makes* a network request as it calls out to ``getRequestHeader``? I may be wrong, but I think we can just read from a client's cookie store instead? Making a network request just so we can get a cookie to make another network request doesn't seem that great.